src/firebase/mock.ts

```diff
@@ -108,53 +117,102 @@ const MockFirebase = new class {
   reset() {
     this.currentUser = new MockUser('test-uid', 'Test User');
     this._docs = {};
-    this._docBindings = {};
+    this._bindings = {};
   }
 
+  // Sets the document at |path| to |data|.
   setDoc(path: string, data: Record<string, any>) {
-    this._docs[path] = data;
-    const bindings = MockFirebase._docBindings[path];
-    if (bindings)
-      bindings.forEach(b => {
-        // TODO: Should probably pass a deep copy here.
-        b.vue.$data[b.name] = data;
-      });
+    this._docs[path] = deepCopy(data);
+
+    // Update all Vue instance data properties bound to the doc.
+    for (const binding of this._bindings[path] || []) {
+      binding.vm.$data[binding.name] = deepCopy(data);
+    }
+  }
+
+  // Returns the document at |path|. Exposed to let tests check stored data.
+  getDoc(path: string) {
+    return this._docs[path];
```

Comment:
Do we need to check if 'path' exists in docs?

Reply:
no, i don't think so. right now, it should return `undefined` for that case, which is the behavior that i'd expect.